pkg/sql/copy_file_upload_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2019 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/sql/tests"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+)
+
+const filename = "/test/test_file_upload.csv"
+
+func writeFile(t *testing.T, testSendFile string, fileContent []byte) {
+	err := os.MkdirAll(filepath.Dir(testSendFile), 0755)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = ioutil.WriteFile(testSendFile, fileContent, 0644)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func runCopyFile(t *testing.T, serverParams base.TestServerArgs, testSendFile string) error {
+	// Make sure we can open this file first
+	reader, err := os.Open(testSendFile)
+	if err != nil {
+		return err
+	}
+
+	s, db, _ := serverutils.StartServer(t, serverParams)
+	defer s.Stopper().Stop(context.TODO())
+
+	if _, err := db.Exec(`
+		CREATE DATABASE d;
+		CREATE TABLE IF NOT EXISTS d.crl_internal_file_upload (b BYTES);
```

Comment:
why does this table need to exist?

Reply:
Oof, good catch this was from a previous version of the prototype, when I was trying things out. Will remove it.